packages/components/src/Typeahead/Typeahead.component.js

```diff
@@ -96,7 +127,7 @@ function Typeahead({ onToggle, icon, position, docked, ...rest }) {
 			onChange: rest.onChange && (event => rest.onChange(event, { value: event.target.value })),
 			onFocus: rest.onFocus,
 			onClick: rest.onClick,
-			onKeyDown: rest.onKeyDown,
+			onKeyDown: manageNavigation ? handleKeyDown : rest.onKeyDown,
```

Comment:
It's not enough to rely on onKeyDown props existence?

Reply:
I wanted to keep the opportunity to add additional behaviours to the handleKey but yes, you're right, it's more elegant like that. I push a fix, thanks